Update documentation about `restore_mode` for valves (#3878)
OttoWinter/esphomedocs: components/valve/template.rst

  requests the valve to be stopped.
- **optimistic** (*Optional*, boolean): Whether to operate in optimistic mode - when in this mode, any command sent to
  the template valve will immediately update the reported state and no lambda needs to be used. Defaults to ``false``.

> - **restore_mode** (*Optional*): Control how the valve attempts to restore state on bootup. Valid options are
>   ``NO_RESTORE``, ``RESTORE`` or ``RESTORE_AND_CALL``. Restore mode ``RESTORE`` attempts to restore the state on
>   startup, but doesn't instruct the valve to return to that state. ``RESTORE_AND_CALL`` additionally instructs the valve
>   to return to the restored state. Defaults to ``NO_RESTORE``

Reviewer: ```suggestion
- **restore_mode** (*Optional*, enum): Control how the valve attempts to restore state on bootup.

  - ``NO_RESTORE`` (Default): Do not save or restore state.
  - ``RESTORE``: Attempts to restore the state on startup, but doesn't instruct the valve to return to that state.
  - ``RESTORE_AND_CALL``: Attempts to restore the state on startup and instructs the valve to return to the restored state.

```
